src/Nix/Value.hs

```diff
@@ -398,66 +400,7 @@ unliftNValue' = hoistNValue' lift
 --
 -- Facts of which are seen below:
 
-
--- | Haskell constant to the Nix constant,
-mkNVConstant' :: Applicative f
-  => NAtom
-  -> NValue' t f m r
-mkNVConstant' = NValue' . pure . NVConstantF
-
--- | Using of Nulls is generally discouraged (in programming language design et al.), but, if you need it.
-nvNull' :: Applicative f
-  => NValue' t f m r
-nvNull' = mkNVConstant' NNull
-
-
--- | Haskell text & context to the Nix text & context,
-mkNVStr' :: Applicative f
-  => NixString
-  -> NValue' t f m r
-mkNVStr' = NValue' . pure . NVStrF
-
-
--- | Haskell @Path@ to the Nix path,
-mkNVPath' :: Applicative f
-  => Path
-  -> NValue' t f m r
-mkNVPath' = NValue' . pure . NVPathF . coerce
-
-
--- | Haskell @[]@ to the Nix @[]@,
-mkNVList' :: Applicative f
-  => [r]
-  -> NValue' t f m r
-mkNVList' = NValue' . pure . NVListF
-
-
 -- | Haskell key-value to the Nix key-value,
-mkNVSet' :: Applicative f
-  => PositionSet
-  -> AttrSet r
-  -> NValue' t f m r
-mkNVSet' p s = NValue' $ pure $ NVSetF p s
-
-
--- | Haskell closure to the Nix closure,
-mkNVClosure' :: (Applicative f, Functor m)
-  => Params ()
-  -> (NValue t f m
-      -> m r
-    )
-  -> NValue' t f m r
-mkNVClosure' x f = NValue' $ pure $ NVClosureF x f
-
-
--- | Haskell functions to the Nix functions!
-mkNVBuiltin' :: (Applicative f, Functor m)
-  => VarName
-  -> (NValue t f m -> m r)
-  -> NValue' t f m r
-mkNVBuiltin' name f = NValue' $ pure $ NVBuiltinF name f
-
-
```

Comment:
This would require at least a note in the changelog.

Currently, through 2 year period of work I understood that we currently can allow low eyeball level of logging (the tunning down of details & preciseness is seen in the [`ChangeLog` itself](https://hackage.haskell.org/package/hnix-0.16.0/changelog)), while the project restructures.
But still, when we are deleting such a big number of functions - the mention of it & how to switch n the log would be good.

Reply:
Agree, should this be added to ChangeLog of 0.17.0 or we should pump up the version again

Reply:
Because changelog already has API breackages in the project, indicated that is would be `0.17.0`.

Skipping versions is not possible (not advised in PVP), skipping major version would mean downstream would need to make a hole in the package ranges and would lead to etc...

Reply:
added